code/datums/components/ntnet_interface.dm

```diff
@@ -1,4 +1,117 @@
 //Thing meant for allowing datums and objects to access an NTnet network datum.
+// we ignore A because of the 24 bit issue
+#define IP_TO_STRING(IP) "[IP.upper>>16].[IP.upper&0xFF].[IP.lower>>16].[IP.lower&0xFF]"
+/datum/ip
+	/// since we don't have access to full 32 bit masks, so deal with 16 bit numbers
+	var/upper = 0
+	var/lower = 0
+	var/text // cached text refrence
+
+/datum/ip/New(A1, A2, A3, A4)
+	if(isnum(A3) && isnum(A4))
+		ASSERT(A1 < 256 && A2 < 256 && A3< 256 && A4< 256)
+		upper = (A1 << 16)  |  A2
+		lower = (A3 << 16)  |  A4
+	else if(istype(A1,/datum/ip))
+		var/datum/ip/A = A1
+		upper = A.upper
+		lower = A.lower
+	else
+		upper = A1
+		lower = A2
+	text = IP_TO_STRING(src)
+
+/datum/ip/proc/to_string()
+	return text
+
+/datum/ip/proc/operator&=(datum/ip/B)
+	upper &= B.upper
+	lower &= B.lower
+	return src
+
+/datum/ip/proc/operator&(datum/ip/B)
+	return new/datum/ip(upper & B.upper,lower & B.lower)
+
+/datum/ip/proc/operator~=(datum/ip/B)
+	return upper == B.upper && lower == B.lower
```

Comment:
Can you make these not use operator overloads? I don't think they particularly help readability, and they definitely make it harder to track down implementations--I can't just alt click a proc name with these.

Reply:
Sure, I am changing how ips work